File: py_results/py_source/question-5-pass-0.py
# Count of quadruples with product of a pair equal to the product of the remaining pair. 
def sameProductQuadruples(nums, N): 
    # Count of quadruples with product of a pair equal to the product of the remaining pair. 
    count = 0
    # Sort the array 
    nums.sort() 
  
    # Traverse the array from left to right 
    for i in range(N): 
        # Traverse the array from right to left 
        for j in range(i + 1, N): 
            # Traverse the array from right to left 
            for k in range(j + 1, N): 
                # Traverse the array from right to left 
                for l in range(k + 1, N): 
                    # If product of a pair is equal to the product of the remaining pair 
                    if (nums[i] * nums[j] == nums[k] * nums[l]): 
                        count += 1
    return count 
  
# Driver code 
nums = [1, 2, 3, 4, 5, 6] 
N = len(nums) 
print("Count of quadruples with product of a pair equal to the product of the remaining pair is", 
      sameProductQuadruples(nums, N)) 
  
# This code is contributed by Nikhil Kumar Singh 

File: cpp_results/cpp_source/question-5-pass-0.cpp
// Count of quadruples with product of a pair equal to the product of the remaining pair. 
 #include <bits/stdc++.h> 
 using namespace std ;
 void sameProductQuadruples ( int nums [ ] , int N ) {
 
    int i, j, k, l ;
    int product ;
    int count = 0 ;
    for ( i = 0 ; i < N ; i++ ) {
        for ( j = i + 1 ; j < N ; j++ ) {
            for ( k = j + 1 ; k < N ; k++ ) {
                for ( l = k + 1 ; l < N ; l++ ) {
                    product = nums [ i ] * nums [ j ] * nums [ k ] * nums [ l ] ;
                    if ( product == nums [ i ] * nums [ j ] * nums [ k ] ) {
                        count++ ;
                    }
                }
            }
        }
    }
    cout << "Count of quadruples with product of a pair equal to the product of the remaining pair is : " << count << endl ;
 }
 int main ( ) {
    int nums [ ] = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 } 